src/__test__/out_object.spec.js

```diff
@@ -28,46 +28,187 @@ describe('Command', () => {
         expect(outObj).toBeInstanceOf(OutObject);
     });
 
-    it('#property() returns a value set by phantom', async () => {
-        let page = await phantom.createPage();
-        let outObj = phantom.createOutObject();
+    describe('#on', () => {
+        'use strict';
+        describe('("onResourceRequested", true)', () => {
+            it('returns a value set by phantom', async () => {
+                let page = await phantom.createPage();
+                let outObj = phantom.createOutObject();
 
-        await page.property('onResourceReceived', function(response, out) {
-            out.lastResponse = response;
-        }, outObj);
+                await page.on('onResourceRequested', true, function(requestData, networkRequest, out) {
```

Comment:
```js
await page.property('onResourceRequested', function(requestData, networkRequest, out) {
                    out.lastRequest = requestData;
                }, outObj);
```
works. So probably `on` and `outobject` have not been tested. 

See my other comment